### ISSUE TITLE
Constants, etc, not being written

### ISSUE THREAD

**Issue by timholy** [ihnorton/Clang.jl#111]
I'm trying to update CUDArt to CUDA6.5. I'm not getting an appreciable number of types, constants, etc being written. Any ideas?

 My (updated) script is:
```julia
using Clang

# The following likely need to be modified for the host system
includes = ["/usr/include",
            "/usr/lib/gcc/x86_64-redhat-linux/4.4.4/include",
            "/usr/lib/gcc/x86_64-linux-gnu/4.8/include",
            "/usr/lib/gcc/x86_64-linux-gnu/4.8/include-fixed"]
cudapath = "/usr/local/cuda-6.5/include"
headers = [joinpath(cudapath,"cuda_runtime_api.h")]

# Customize the writing of functions
skip_error_check = [:cudaStreamQuery,:cudaGetLastError,:cudaPeekAtLastError]
function rewriter(ex::Expr)
    ex.head == :function || return ex
    decl, body = ex.args[1], ex.args[2]
    # omit types from function prototypes
    for i = 2:length(decl.args)
        a = decl.args[i]
        if a.head == :(::)
            decl.args[i] = a.args[1]
        end
    end
    # Error check functions that return a cudaError_t (with exceptions)
    ccallexpr = body.args[1]
    if ccallexpr.head != :ccall
        error("Unexpected body expression: ", body)
    end
    rettype = ccallexpr.args[2]
    if rettype == :cudaError_t
        fname = decl.args[1]
        if !in(fname, skip_error_check)
            body.args[1] = Expr(:call, :checkerror, deepcopy(ccallexpr))
        end
    end
    ex
end

rewriter(A::Array) = [rewriter(a) for a in A]

context=wrap_c.init(output_file="gen_libcudart.jl",
                    common_file="gen_libcudart_h.jl",
                    header_library=x->"libcudart",
                    clang_includes=includes,
                    clang_diagnostics=true,
#                    header_wrapped=(x,y)->contains(x,"cuda"),
                    header_wrapped=(x,y)->true,
                    rewriter=rewriter)

context.options = wrap_c.InternalOptions(true,true)  # wrap structs, too

# Execute the wrap
wrap_c.wrap_c_headers(context, headers)
```
When I run this, the output is
```
julia> include("wrap_cuda.jl")
WARNING: wrap_c_headers: deprecated
WRAPPING HEADER: /usr/local/cuda-6.5/include/cuda_runtime_api.h
WARNING: Not wrapping Clang.cindex.InclusionDirective   host_defines.h
WARNING: Not wrapping Clang.cindex.InclusionDirective   builtin_types.h
WARNING: Not wrapping Clang.cindex.InclusionDirective   cuda_device_runtime_api.h
WARNING: Not wrapping Clang.cindex.MacroInstantiation   CUDARTAPI
WARNING: Not wrapping Clang.cindex.MacroInstantiation   CUDARTAPI
WARNING: Not wrapping Clang.cindex.MacroInstantiation   CUDARTAPI
WARNING: Not wrapping Clang.cindex.MacroInstantiation   CUDARTAPI
...
WARNING: Not wrapping Clang.cindex.MacroInstantiation   CUDARTAPI
WARNING: Not wrapping Clang.cindex.MacroInstantiation   CUDARTAPI
writing gen_libcudart.jl
1-element Array{Any,1}:
 nothing

julia> 
```
The contents of `gen_libcudart.jl` look, upon casual inspection, just fine. The entire contents of `gen_libcudart_h.jl` are:
```julia
const CUDART_VERSION = 6050
const CUDART_DEVICE = __device__

typealias cudaStreamCallback_t Ptr{Void}
```
whereas when I did this before with CUDA5.0, I got
```julia
const cudaHostAllocDefault = 0x00
const cudaHostAllocPortable = 0x01
const cudaHostAllocMapped = 0x02
const cudaHostAllocWriteCombined = 0x04
const cudaHostRegisterDefault = 0x00
... (987 lines in total)
```

**Comment by timholy** [comment]
I should have said, I'm running this on julia 0.4. I'll test julia 0.3, too (have to build with LLVM_CLANG=1 first, so it will be a while).

**Comment by timholy** [comment]
Also an issue on julia 0.3, using the master branch.

**Comment by timholy** [comment]
OK, I believe I've fixed all of the most pressing bugs. I did find that I had to explicitly list the headers I wanted: by using
```julia
headers = ["cuda_runtime_api.h", "driver_types.h", "vector_types.h"]
headers = [joinpath(cudapath,h) for h in headers]
```
I could get it to generate a common file that seems OK.

I'm not sure if it's intended that I have to manually list those files. In particular, I can't find any evidence that `wrap_c.wrap_header` makes use of the `deps` that get stored in an `ExprUnit`. So the types needed by functions are skipped over due to [this line](https://github.com/ihnorton/Clang.jl/blob/549828212f9d418b037b5b44c71723adc38880d9/src/wrap_c.jl#L625).
